### DISCUSSION TITLE
Menubar

### DISCUSSION THREAD

**Discussion by SeleDreams** [cycfi/elements#341] [Elements]
Hi, I was wondering if elements has a widget relative to top menus ? (the ones for "file" "edit" "help" etc)
because I tried to make one atm using standard menu buttons but it doesn't feel as good as the usual top menus because for instance moving the mouse won't make switch to the next dropdown and they also don't really feel connected, also uually in top menus there aren't down arrows
![Capture d’écran du 2022-09-22 18-37-30](https://user-images.githubusercontent.com/16335601/191803615-fafcf802-4382-4f5f-9dff-cb637a5a8f79.png)

**Comment by redtide** [comment]
I'm afraid not. What you are asking for is a menubar, which in some systems like macOS is managed globally so not part of the main window (so I wonder if it will ever be implemented should it be something native instead).

**Comment by djowel** [comment]
OTOH, you can change the look of the menu button quite easily. Just tweak the menu button background.

**Comment by djowel** [comment]
Hmmm... I think it's better to have a real menubar as an element. I see what you are saying. Should be relatively easy to do.

**Comment by SeleDreams** [comment]
> Hmmm... I think it's better to have a real menubar as an element. I see what you are saying. Should be relatively easy to do.

when you say real menubar, do you mean an OS based menubar or an elements specific one ?

**Comment by djowel** [comment]
> > Hmmm... I think it's better to have a real menubar as an element. I see what you are saying. Should be relatively easy to do.
> 
> when you say real menubar, do you mean an OS based menubar or an elements specific one ?

Elements specific.

**Comment by SeleDreams** [comment]
I see, that's good too as it would make it more portable, if we want an OS specific one like a macos one for instance i think it's still possible to disable the elements one at runtime and enable the OSX one using the osx api
